### ISSUE TITLE
How to modify the loss to encourage exploration?

### ISSUE THREAD

**Issue by jiazou-bigdata** [hongzimao/pensieve#44]
I'm trying to apply a3c.py to a different problem. However it quickly converge to the dominant action and never move again. My question is: whether there are any suggestions for modifying parameters or loss function to encourage exploration? Have you tried that in pensieve?  Thanks!

**Comment by hongzimao** [comment]
You might find this README useful: https://github.com/hongzimao/pensieve/blob/master/sim/README.md

> As reported by the A3C paper (http://proceedings.mlr.press/v48/mniha16.pdf) and a faithful implementation (https://openreview.net/pdf?id=Hk3mPK5gg), we also found the exploration factor in the actor network quite crucial for achieving good performance. A general strategy to train our system is to first set `ENTROPY_WEIGHT` in `a3c.py` to be a large value (in the scale of 1 to 5) in the beginning, then gradually reduce the value to `0.1` (after at least 100,000 iterations).

**Comment by jiazou-bigdata** [comment]
Thanks a lot Hongzi! 
I actually read your paper and set ENTROPY_WEIGHT to 7 (should it must be no larger than 5?) But I still feel it does insufficient exploration. I also observed the gradients of critic network are all zero. I changed to leaky_relu, but the critic_gradient is still all zero. Have you experienced that in Pensieve? Any suggestions?

Best Regards,
Jia

**Comment by hongzimao** [comment]
That's weird. I don't think we had this issue before. Can you see if the loss of value network goes down?

**Comment by jiazou-bigdata** [comment]
In my last run, td_loss goes up from 26.xxx to 30.xxx and then drops to 26.xxx. I can rerun it and send a complete log of td_loss tomorrow. Thanks!

Best Regards,
Jia

**Comment by jiazou-bigdata** [comment]
Problem solved, it is because my neural network architecture is not so reasonable. Now it has been corrected, and I can see obvious exploration and critic gradients are not zero any more.
